lesson03.md

```diff
@@ -0,0 +1,324 @@
+![Variable](images/conditionals_red_or_blue.jpg)<!-- .element width="400px" -->
+
+A conditional statement __decides__ whether specific code runs based on whether a condition is true or false.
+<!-- .element: class="fragment" -->
+
+---
+
+### if statements
+
+`if` statement specifies code to be executed if a condition is true.
+
+```js
+if (condition) {
+    // run code if condition is true
+}
+```
+
+- The condition must be a boolean.
+- The code in the block is executed only if the boolean is true.
+- If the boolean is false, the code block is skipped.
+
+---
+
+### Testing conditions using Comparision operators
+
+Comparison operators are used to test if a condition is true or false so we can decide when to run certain code.
+
+```js
+if (sum === 30) {
+  document.body.textContent = 'sum is equal to 30';
+}
+```
+<!-- .element: class="fragment" -->
+
+---
+
+### if Statements - Examples
+
+```js
+if (food === "pizza") {
+  document.body.textContent = "My favorite food is pizza!";
+}
+```
+
+```js
+if (temperature > 20) {
+  document.body.textContent = "Yay, it's warm again!";
+}
+```
+
+```js
+if (canSpeakFrench) {
+  // same as: canSpeakFrench === true
+  document.body.textContent = "Bonjour!";
+}
+```
+
+---
+
+### Code Blocks: {}
+
+Code blocks contain the code between `{` and `}`, and should be indented for better readability:
+
+```js
+if (language === "German") {
+  // In VSCode, you can indent code by pressing the "Tab" key,
+  // or right-click and choose "Format Document"
+  document.body.textContent = "Guten Tag";
+}
+```
+
+---
+
+### Code Blocks: {}
+
+All variables defined in code blocks only exist inside the block (technical term: “block scope”)
+
+```js
+let name = "John Doe";
+if (name === "John Doe") {
+  let greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  let greeting = "Hello my dear";
+}
+document.body.textContent = greeting; // ERROR!
+```
+
+---
+
+### Code Blocks: {}
+
+Correct solution:
+
+```js
+let name = "John Doe";
+let greeting;
+if (name === "John Doe") {
+  greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  greeting = "Hello my dear";
+}
+document.body.textContent = greeting; // "Hi dude"
+```
+
+---
+
+## Logical operators: AND, OR and NOT
+
+Logical operators are helpful if you want to test multiple conditions in an `if` statement.
+
+- `&&` (AND) – All conditions must be true for the statement to execute.
+
+<!-- .element: class="fragment" -->
+
+- `||` (OR) – At least one condition must be true for the statement to execute.
+
+<!-- .element: class="fragment" -->
+
+- `!` (NOT) – Reverses the truth value of a condition (true becomes false and vice versa).
+
+<!-- .element: class="fragment" -->
+
+---
+
+## Logical operators - Examples
+
+`&&` (AND)
+```js
+let age = 25;
+let hasLicense = true;
+if (age >= 18 && hasLicense) {
+    console.log("You can drive!");
+}
+```
+
+`||`  (OR)
+```js
+let isRaining = false;
+let hasUmbrella = true;
+if (isRaining || hasUmbrella) {
```

Comment:
I just realized that this example
makes no sense :) We can go outside if it's raining and we habe no Umbrella?